src/Stripe.net/Entities/SubscriptionsSchedules/SubscriptionSchedule.cs

```diff
@@ -0,0 +1,179 @@
+namespace Stripe
+{
+    using System;
+    using System.Collections.Generic;
+    using Newtonsoft.Json;
+    using Stripe.Infrastructure;
+
+    public class SubscriptionSchedule : StripeEntity, IHasId, IHasMetadata, IHasObject
+    {
+        /// <summary>
+        /// Unique identifier for the object.
+        /// </summary>
+        [JsonProperty("id")]
+        public string Id { get; set; }
+
+        /// <summary>
+        /// String representing the object’s type. Objects of the same type share the same value.
+        /// </summary>
+        [JsonProperty("object")]
+        public string Object { get; set; }
+
+        /// <summary>
+        /// One of <see cref="Billing" />. When charging automatically, Stripe will attempt to pay
+        /// this subscription at the end of the cycle using the default source attached to the
+        /// customer. When sending an invoice, Stripe will email your customer an invoice with
+        /// payment instructions.
+        /// </summary>
+        [JsonProperty("billing")]
+        public Billing? Billing { get; set; }
+
+        /// <summary>
+        /// Define thresholds at which an invoice will be sent, and the subscription advanced to a
+        /// new billing period
+        /// </summary>
+        [JsonProperty("billing_thresholds")]
+        public SubscriptionBillingThresholds BillingThresholds { get; set; }
+
+        /// <summary>
+        /// Time at which the subscription schedule was canceled. Measured in seconds since the
+        /// Unix epoch.
+        /// </summary>
+        [JsonProperty("canceled_at")]
+        [JsonConverter(typeof(DateTimeConverter))]
+        public DateTime? CanceledAt { get; set; }
+
+        /// <summary>
+        /// Time at which the subscription schedule was completed. Measured in seconds since the
+        /// Unix epoch.
+        /// </summary>
+        [JsonProperty("completed_at")]
+        [JsonConverter(typeof(DateTimeConverter))]
+        public DateTime? CompletedAt { get; set; }
+
+        /// <summary>
+        /// Time at which the object was created. Measured in seconds since the Unix epoch.
+        /// </summary>
+        [JsonProperty("created")]
+        [JsonConverter(typeof(DateTimeConverter))]
+        public DateTime? Created { get; set; }
+
+        /// <summary>
+        /// Object representing the start and end dates for the current phase of the subscription
+        /// schedule, if it is <code>active</code>.
+        /// </summary>
+        [JsonProperty("current_phase")]
+        public SubscriptionScheduleCurrentPhase CurrentPhase { get; set; }
+
+        // TODO: figure out how we document those
```

Comment:
Should I bother documenting since this is going away in your next major version?

Reply:
@ob-stripe Sure but the entire "region" is going away though. So my todo/question holds: How should I document this? Can you show me an example with this region on where to put the comments so that your rebase is not too painful?

Reply:
The region would stay there too, what would change is:
- `public string CustomerId { get; set; }` --> `public string CustomerId => InternalCustomer.Id;`
- `public Customer Customer { get; set; }` --> `public Customer Customer => InternalCustomer.ExpandedObject;`
- `internal object InternalCustomer { ... }` --> `internal ExpandableField<Customer> InternalCustomer;`

Here's an example for documenting the `public` accessors for the ID and object: https://github.com/stripe/stripe-dotnet/blob/master/src/Stripe.net/Entities/Disputes/Evidence.cs#L16-L28. Not sure it's the best format, but it's something.